ecs-agent/csiclient/csi_client.go

```diff
@@ -44,31 +57,83 @@ type csiClient struct {
 }
 
 // NewCSIClient creates a CSI client for the communication with CSI driver daemon.
-func NewCSIClient(socketIn string) CSIClient {
-	return &csiClient{csiSocket: socketIn}
+func NewCSIClient(socketIn string) csiClient {
+	return csiClient{csiSocket: socketIn}
 }
 
-// GetVolumeMetrics returns volume usage.
-func (cc *csiClient) GetVolumeMetrics(volumeId string, hostMountPath string) (*Metrics, error) {
-	// Set up a connection to the server
-	dialer := func(addr string, t time.Duration) (net.Conn, error) {
-		return net.Dial(PROTOCOL, addr)
+func (cc *csiClient) NodeStageVolume(ctx context.Context,
+	volID string,
+	publishContext map[string]string,
+	stagingTargetPath string,
+	fsType string,
+	accessMode v1.PersistentVolumeAccessMode,
+	secrets map[string]string,
+	volumeContext map[string]string,
+	mountOptions []string,
+	fsGroup *int64,
+) error {
+	conn, err := cc.grpcDialConnect()
+	if err != nil {
+		logger.Error("NodeStage: CSI Connection Error")
```

Comment:
Can we log the error as well? Or to avoid to log same error in couple places, can we just wrap the error with some extra info?